pandas/core/indexes/extension.py

```diff
@@ -223,29 +219,14 @@ def __getitem__(self, key):
         deprecate_ndim_indexing(result)
         return result
 
-    def __iter__(self):
-        return self._data.__iter__()
```

Comment:
Should be equivalent to 

https://github.com/pandas-dev/pandas/blob/9c08fe1e192ed4ab197b042d8477f583d7a1cdeb/pandas/core/base.py#L1034-L1051

Reply:
Sure (alternatively could remove L1047-1049 from the base class implementation.

Reply:
> alternatively could remove L1047-1049 from the base class implementation

No, because Series also uses that